src/IO/H5/VolumeData.hpp

```diff
@@ -136,6 +136,20 @@ class VolumeData : public h5::Object {
   /// `observation_id`
   std::vector<std::vector<size_t>> get_extents(size_t observation_id) const;
 
+  /// Retrieve volume data for IDs in
+  /// \f$[start_observation_value, end_observation_value]\f$.
+  ///
+  /// - If `start_observation_value` is not `std::nullopt` then return
+  ///   everything from the beginning of the data to `end_observation_value`. If
+  ///   `end_observation_value` is `std::nullopt` then return everything from
+  ///   `start_observation_value` to the end of the data.
+  auto get_data_by_element(std::optional<double> start_observation_value,
```

Comment:
please document what is returned (the return type is not showing up in the detailed description).  Also nice to know what the tuple types correspond to.   Also note how everything is sorted.